src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/ConfigurationTest.php

```diff
@@ -607,6 +607,30 @@ public function testScopedHttpClientsInheritRateLimiterAndRetryFailedConfigurati
         $this->assertSame(999, $scopedClients['qux']['retry_failed']['delay']);
     }
 
+    public function testSerializerJsonDetailedErrorMessagesEnabledByDefaultWithDebugEnabled()
+    {
+        $processor = new Processor();
+        $config = $processor->processConfiguration(new Configuration(true), [
+            [
+                'serializer' => null,
+            ],
+        ]);
+
+        $this->assertSame([JsonDecode::DETAILED_ERROR_MESSAGES => true], $config['serializer']['default_context'] ?? []);
```

Comment:
I find it weird to fallback on an empty array in case of something being wrong with `$config['serializer']['default_context']`. Are we not in total control of what's happening in the test?

Reply:
I don't want the test to emit PHP errors (warnings/notices) in case something with the tested code does not work as expected (for example the `serializer` key not being present).

Reply:
Ah yes, optional dependencies can be tricky indeed